js/views/modals/purchase/Purchase.js

```diff
@@ -278,6 +281,19 @@ export default class extends BaseModal {
     };
   }
 
+  setState(state = {}, options = {}) {
+    const superReturn = super.setState(state, options);
+
+    if (
+      this.receipt &&
+      this.getState().phase !== 'pay'
+    ) {
+      this.receipt.showTotalTip = false;
+    }
+
+    return superReturn;
```

Comment:
Would it be simpler to just return super.setState(state, options) here instead of setting it as superReturn first?

Reply:
That's super minor, so up to you if you want to change it. Either way we can merge this.

Reply:
The reason it's done this way is so the `this.getState().phase !== 'pay'` check happens after the super call changes state. So we're basically checking if the resulting phase is not 'pay', we hide that tooltip.

I suppose I could have just checked if the passed in state.phase !== 'pay', but it feels like maybe there are or could be some edge cases if the state were somehow changed a different way, e.g. maybe someone sets the initial state to not be 'pay'.

Reply:
Ah, ok, that makes sense. Thanks!